build.gradle

```diff
@@ -124,39 +139,197 @@ publishing {
             }
         }
     }
+    // Configure the staging directory to put all artifacts into prior to publishing to Maven
+    // Central. To publish to a Maven repository, JReleaser must be configured to point to a staging
+    // directory that contains all artifacts in Maven's default layout.
+    repositories {
+        maven {
+            url = stagingDeployDir
+        }
+    }
 }
 
-// Maven Central requires signed artifacts.
-signing {
-    def signingKey = findProperty("signingKey")
-    def signingPassword = findProperty("signingPassword")
-    useInMemoryPgpKeys(signingKey, signingPassword)
-    sign publishing.publications.mavenJava
+// -----------------------------------------------------------------------------
+// CONFIGURATION: org.jreleaser
+// -----------------------------------------------------------------------------
+
+// JReleaser plugin configuration.
+jreleaser {
+    // NOTE: Config is applied in this order: inline > env var.
+
+    // Artifact signing configuration.
+    // See https://jreleaser.org/guide/latest/reference/signing.html
+    signing {
+        // Enables or disables file signing.
+        // Supported values are [`NEVER`, `ALWAYS`, `RELEASE`, `SNAPSHOT`].
+        // Defaults to `NEVER`.
+        active = 'ALWAYS'
+
+        // Generates an armored signature.
+        // Defaults to `false`.
+        //
+        // We set this to `true` to produce human-readable `.asc` signature files.
+        // NOTE: Signing fails with "Unexpected error" when set to `false`.
+        armored = true
+
+        // Verify signature files.
+        // If `false` then `publicKey` may be omitted.
+        // Defaults to `true`.
+        //
+        // NOTE: Disabled because there's no need to verify signatures immediately after signing.
+        // If we want to enable this in the future, we'll need to configure `publicKey` to our PGP
+        // public key.
+        verify = false
+
+        // The passphrase required to read secret keys.
+        // Set via env var `JRELEASER_GPG_PASSPHRASE`.
+        //
+        // NOTE: We set this using a project property. See
+        // https://docs.gradle.org/current/userguide/build_environment.html#sec:project_properties
+        // NOTE: This raises an error if left blank, however this is a bug that should be fixed in
+        // the 1.19.0 release. See https://github.com/jreleaser/jreleaser/issues/1883.
+        passphrase = findProperty("signingPassword") ?: '__DUMMY_VALUE__'
```

Comment:
Why are we using properties instead of env vars?
I can't tell based on the linked article.

Reply:
@cameron-thomson-bsl responded on ticket:
https://brightsparklabs.atlassian.net/browse/RAD-233?focusedCommentId=61000